src/components/TextInput/BaseTextInput/index.tsx

```diff
@@ -77,72 +113,66 @@ function BaseTextInput(props) {
     );
 
     const activateLabel = useCallback(() => {
-        const value = props.value || '';
+        const newValue = value ?? '';
 
-        if (value.length < 0 || isLabelActive.current) {
+        if (newValue.length < 0 || isLabelActive.current) {
             return;
         }
 
         animateLabel(styleConst.ACTIVE_LABEL_TRANSLATE_Y, styleConst.ACTIVE_LABEL_SCALE);
         isLabelActive.current = true;
-    }, [animateLabel, props.value]);
+    }, [animateLabel, value]);
 
     const deactivateLabel = useCallback(() => {
-        const value = props.value || '';
+        const newValue = value ?? '';
 
-        if (props.forceActiveLabel || value.length !== 0 || props.prefixCharacter) {
+        if (!!forceActiveLabel || newValue.length !== 0 || prefixCharacter) {
             return;
         }
 
         animateLabel(styleConst.INACTIVE_LABEL_TRANSLATE_Y, styleConst.INACTIVE_LABEL_SCALE);
         isLabelActive.current = false;
-    }, [animateLabel, props.forceActiveLabel, props.prefixCharacter, props.value]);
+    }, [animateLabel, forceActiveLabel, prefixCharacter, value]);
 
-    const onFocus = (event) => {
-        if (props.onFocus) {
-            props.onFocus(event);
-        }
+    const onFocus = (event: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        inputProps.onFocus?.(event);
         setIsFocused(true);
     };
 
-    const onBlur = (event) => {
-        if (props.onBlur) {
-            props.onBlur(event);
-        }
+    const onBlur = (event: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        inputProps.onBlur?.(event);
         setIsFocused(false);
     };
 
-    const onPress = (event) => {
-        if (props.disabled) {
+    const onPress = (event?: GestureResponderEvent | KeyboardEvent) => {
+        if (!!inputProps.disabled || !event) {
             return;
         }
 
-        if (props.onPress) {
-            props.onPress(event);
-        }
+        inputProps.onPress?.(event);
 
-        if (!event.isDefaultPrevented()) {
-            input.current.focus();
+        if ('isDefaultPrevented' in event && !event?.isDefaultPrevented()) {
```

Comment:
I feel like you've done this in lots of places, and forgive my ts-noobness please :D But why do we have to check `'isDefaultPrevented' in event` AND also use optional chaining `event?.isDefaultPrevented`?

My guess is:
1. `'isDefaultPrevented' in event` is to satisfy typescript
2. optional chaining is to make sure `isDefaultPrevented` is not undefined / null